TT_REST_API/FillDownload 2.0/python/fill_download.py

```diff
@@ -0,0 +1,932 @@
+        try:
+            login_info = api_request(url=token_url, headers=ttid_headers,
+                                      data=ttid_data, http_method='post')
+        except AssertionError:
+            stop_running.set()
+            raise
+
+        token = '{} {}'.format(login_info['token_type'].capitalize(),
+                               login_info['access_token'])
+
+        token_expiry = time.time() + login_info['seconds_until_expiry']
+
+        with lock:
+            common.api_http_header = {'x-api-key': key, 'Authorization': token}
+
+        while time.time() < token_expiry-10 and not stop_running.is_set():
+            time.sleep(1)
+
+def build_enums(environment, headers):
+    # Order Data Enums
+    order_data_download_url = '{}/ttledger/{}/orderdata'.format(TT_URL_BASE, environment)
+    fill_data = api_request(order_data_download_url, headers)
+    enums = fill_data['orderData']
+
+    # Market Name Enums
+    markets_download_url = '{}/ttpds/{}/markets'.format(TT_URL_BASE, environment)
+    response = api_request(markets_download_url, headers)
+    enums['markets'] = {info['id']: info['name'] for info in response['markets']}
+
+    # Market Code Enums
+    mics_download_url = '{}/ttpds/{}/mics'.format(TT_URL_BASE, environment)
+    response = api_request(mics_download_url, headers)
+    enums['mics'] = {}
+    for mic in response['markets']:
+        if mic['desc']:
+            enums['mics'][mic['id']] = mic['desc']
+        elif mic['marketId'] in enums['markets']:
+            enums['mics'][int(mic['id'])] = \
+                '{}_All'.format(enums['markets'][mic['marketId']])
+
+    # Product Data Enums
+    prod_data_download_url = '{}/ttpds/{}/productdata'.format(TT_URL_BASE, environment)
+    prod_data = api_request(prod_data_download_url, headers)
+    for enum in prod_data:
+        if enum in ('status', 'message', 'lastPage'):
+            continue
+        enums[enum] = {int(info['id']): info['name'] for info in prod_data[enum]}
+
+    # Instrument Date Enums
+    inst_data_download_url = '{}/ttpds/{}/instrumentdata'.format(TT_URL_BASE, environment)
+    inst_data = api_request(inst_data_download_url, headers)
+    for enum in inst_data:
+        if enum in ('status', 'message', 'lastPage'):
+            continue
+        enum_name = 'desc' if 'desc' in inst_data[enum][0] else 'name'
+        enums[enum] = {int(info['id']): info[enum_name] for info in inst_data[enum]}
+
+    return enums
+
+
+def retrieve_fills(environment, headers, min_time_stamp=None, max_time_stamp=None):
+    fill_download_url = '{}/ttledger/{}/fills'.format(TT_URL_BASE, environment)
+    if min_time_stamp:
+        fill_download_url += '?minTimestamp={}'.format(min_time_stamp)
+    if max_time_stamp:
+        fill_download_url += '?maxTimestamp={}'.format(max_time_stamp)
+    fills = api_request(fill_download_url, headers, request_timeout=True)
+    # if fills.status_code == 408:
+    if fills['status'] != 'Ok':
```

Comment:
This code will fail it the actual status is 408 because the function api_request will return the raw response which I think doesn't have a 'status' but a 'status_code'. Can you please check this.